nrpy/helpers/expr_tree.py

```diff
@@ -190,6 +190,31 @@ def __repr__(self) -> str:
     __str__ = __repr__
 
 
+def get_unique_expression_symbols(
+    expr: sp.Expr, exclude: Union[List[str], None] = None
+) -> List[str]:
+    """
+    Get a unique list of expression symbols.
+
+    :param expr: Sympy expression
+    :param exclude: List of symbol names to exclude
+    :returns: List of unique symbols from the expression
+
+    DOCTEST:
+    >>> from sympy.abc import a, b
+    >>> from sympy import cos, sin
+    >>> xx0 = sp.Symbol('xx0')
+    >>> x = cos(a + b)**2 ++ xx0
+    >>> get_unique_expression_symbols(x, ["xx0"])
+    ['a', 'b']
+    """
+    if exclude is None:
+        exclude = []
+
+    symbols = {str(sym) for sym in expr.free_symbols}
+    return sorted([sym for sym in symbols if not sym in exclude])
```

Comment:
I don't like that we're converting the symbols to strings here, as it's a recipe for nonrobustness (and slowness too!) Why can't exclude be a list of symbols?

Reply:
In many places, the inclusion of `xx0`, `xx1`, `xx2` is hard coded in function bodies, thus the current use, e.g., is the following:
`unique_symbols += get_unique_expression_symbols(expr, exclude=[f"xx{i}" for i in range(3)])`

Unfortunately this cannot be changed to something like:
`unique_symbols += get_unique_expression_symbols(expr, exclude=[sp.Symbol(f"xx{i}") for i in range(3)])`

as these are unique Sympy objects and thus cannot be compared.

Reply:
Actually, I think the latter (comparing symbols) *will* work! You may need to ensure that real=True is set properly.

Reply:
`expr_tree.py` is no longer modified.  I have moved this functionality `nrpy/helpers/expression_utils.py` per our slack discussion after updating the internal algorithm and function name.